test/jdk/java/lang/ProcessBuilder/Basic.java

```diff
@@ -2647,14 +2613,63 @@ public void run() {
 
             long end = System.nanoTime();
             if ((end - start) < 500000000)
-                fail("Test failed: waitFor didn't take long enough");
+                fail("Test failed: waitFor didn't take long enough (" + (end - start) + "ns)");
 
             p.destroy();
 
             p.waitFor(1000, TimeUnit.MILLISECONDS);
         } catch (Throwable t) { unexpected(t); }
     }
 
+    // Path to native executables, if any
+    private static final String TEST_NATIVEPATH = System.getProperty("test.nativepath");
+
+    // Path where "sleep" program may be found" or null
+    private static final Path SLEEP_PATH = initSleepPath();
+
+    /**
+     * Compute the Path to a sleep executable.
+     * @return a Path to sleep or BasicSleep(.exe) or null if none
+     */
+    private static Path initSleepPath() {
+        if (Windows.is() && TEST_NATIVEPATH != null) {
+            // exeBasicSleep is equivalent to sleep on Unix
+            Path exePath = Path.of(TEST_NATIVEPATH).resolve("BasicSleep.exe");
+            if (exePath.toFile().canExecute()) {
+                return exePath;
+            }
+        }
+
+        Path exePath = Path.of("/bin").resolve("sleep");
+        if (exePath.toFile().canExecute()) {
+            return exePath;
+        }
+        return null;
+    }
+
+    /**
+     * Return the list of process arguments for a child to sleep 10 minutes (600 seconds).
+     *
+     * @return A list of process arguments to sleep 10 minutes.
+     */
+    private static List<String> getSleepArgs() {
+        List<String> childArgs = null;
+        if (SLEEP_PATH != null) {
+            childArgs = List.of(SLEEP_PATH.toString(), "600");
+        } else {
+            // Fallback to the JavaChild sleep does a sleep for 10 minutes.
+            // The fallback the Java$Child is used if the test is run without building
+            // the BasicSleep native executable (for Windows).
+
+            Path exePath = Path.of(TEST_NATIVEPATH).resolve("BasicSleep.exe");
+            System.out.println("exePath: " + exePath + ", canExec: " + exePath.toFile().canExecute());
```

Comment:
What is this for??

Reply:
Ioi was concerned that the fallback to Java would be used silently and the old intermittent issue would return.

Reply:
Well Ioi asked for a comment explaining why the fallback was necessary, but the println doesn't explain anything. I assume it should say something like "Unable to find ... falling back to Java version"?